File: C/gnomesort/deepseek-r1-distill-llama-70b.c
#include <stdio.h>

void gnome_sort_optimized(int arr[], int size) {
    int swapped;
    do {
        swapped = 0;
        for (int i = 1; i < size; ) {
            if (arr[i] > arr[i-1]) {
                ++i;
            } else {
                // Swap elements
                int temp = arr[i];
                arr[i] = arr[i-1];
                arr[i-1] = temp;
                swapped = 1;
                if (i > 1) {
                    --i;
                } else {
                    i = 1;
                }
            }
        }
    } while (swapped);
}

int main() {
    int arr[1000];
    int size = 0;

    FILE *file = fopen("../../inputs.txt", "r");
    if (!file) {
        return 1;
    }

    while (fscanf(file, "%d", &arr[size]) == 1 && size < 1000) {
        size++;
    }
    fclose(file);

    gnome_sort_optimized(arr, size);

    for (int i = 0; i < size; ++i) {
        printf("%d ", arr[i]);
    }
    printf("\n");
    return 0;
}

File: Python/gnomesort/deepseek-r1-distill-llama-70b.py
def gnome_sort(arr):
    last_swap_pos = -1
    i = 0
    n = len(arr)
    while i < n - 1:
        if arr[i] > arr[i + 1]:
            arr[i], arr[i + 1] = arr[i + 1], arr[i]
            last_swap_pos = i
            i = max(0, i - 1)
        else:
            if last_swap_pos != -1 and i > last_swap_pos:
                last_swap_pos = -1
            i += 1
    return arr

# Example usage with 1000 random inputs (example with 5 elements for demonstration)
import sys

def main():
    input_path = "inputs.txt" if len(sys.argv) < 2 else sys.argv[1]

    try:
        with open(input_path, 'r') as f:
            arr = [int(line.strip()) for line in f if line.strip()]
    except FileNotFoundError:
        return

    sorted_arr = gnome_sort(arr)

if __name__ == "__main__":
    main()
